micrometer-core/src/main/java/io/micrometer/core/instrument/composite/CompositeCustomMeter.java

```diff
@@ -20,22 +20,69 @@
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.internal.DefaultMeter;
 
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
 class CompositeCustomMeter extends DefaultMeter implements CompositeMeter {
+    private AtomicBoolean childrenGuard = new AtomicBoolean();
+    private Map<MeterRegistry, Meter> children = Collections.emptyMap();
```

Comment:
With the additional `remove` method made available in #2362, I think the changes to `CompositeCustomMeter` can be simplified to doing `registry.remove(getId(), true)` in the `remove` method here. I don't think we need to track the child registries like in `AbstractCompositeMeter` because we don't need to propagate incrememnts/recording to children using the `forEachChild` method.
What do you think?